readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -115,32 +115,22 @@ def get_config_params(self):
         downloads = []
         subproject_urls = []
         # Avoid hitting database and API if using Docker build environment
-        if settings.DONT_HIT_API:
-            if self.project.has_feature(Feature.ALL_VERSIONS_IN_HTML_CONTEXT):
+        if self.project.has_feature(Feature.ALL_VERSIONS_IN_HTML_CONTEXT):
+            if settings.DONT_HIT_API:
                 versions = self.project.active_versions()
-            else:
-                versions = self.project.active_versions().filter(
-                    privacy_level=PUBLIC,
-                )
-            downloads = self.version.get_downloads(pretty=True)
-            subproject_urls = self.project.get_subproject_urls()
-        else:
-            try:
-                versions = self.project.api_versions()
-                if not self.project.has_feature(Feature.ALL_VERSIONS_IN_HTML_CONTEXT):
-                    versions = [
-                        v
-                        for v in versions
-                        if v.privacy_level == PUBLIC
```

Comment:
Let me know if you still want to show only public versions if a project has the feature enabled.